apps/fishing-map/features/vessel/activity/activity-by-type/activity-by-type.hook.ts

```diff
@@ -0,0 +1,68 @@
+import { useCallback, useMemo, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { EventType, EventTypes } from '@globalfishingwatch/api-types'
+//import { selectEventResourcesLoading } from 'features/resources/resources.selectors'
+import { RenderedEvent } from 'features/vessel/activity/vessels-activity.selectors'
+import { EventGroup } from 'types/activity'
+import { selectFilteredEventsWithMainPortVisit } from './activity-by-type.selectors'
+
+const calculateQuantity = {
+  [EventTypes.Encounter]: (events: RenderedEvent[]) => events.length ?? 0,
+  [EventTypes.Fishing]: (events: RenderedEvent[]) =>
+    events?.reduce((p, c) => p + c.duration ?? 0, 0) ?? 0,
+  [EventTypes.Loitering]: (events: RenderedEvent[]) => events.length ?? 0,
+  [EventTypes.Port]: (events: RenderedEvent[]) =>
+    Math.ceil(events.filter((event) => !event?.subEvent).length ?? 0),
+  [EventTypes.Gap]: (events: RenderedEvent[]) => events.length ?? 0,
+}
+
+export const useActivityByType = () => {
+  //const eventsLoading = useSelector(selectEventResourcesLoading)
+  const eventsList = useSelector(selectFilteredEventsWithMainPortVisit)
+  const [expandedGroups, setExpandedGroups] = useState<EventType[]>([])
+  console.log(eventsList)
+  const eventsByType = useMemo(
```

Comment:
I want all the grouping and filtering logic in the selector and not spread in the app. Also, this is not performant at all.
Could you take a look?